RNA-seq/05-nb_cell_line_DESeq2.Rmd

```diff
@@ -154,124 +169,131 @@ Let's save this to our results file.
+           color = logmean)) +
+  geom_point(alpha = 0.1) +
+  theme_bw() +
+  scale_color_viridis_c() + 
+  coord_cartesian(xlim = c(-10,10), ylim = c(-10,10)) # zoom in on the middle
 ```
 
-Now we will use `signif_genes` to filter `vst_data` to only have significant genes' data. 
+We will now do a bit of manipulation to store the results in a data frame for easier manipulation.
 
-```{r signif_vst_data}
-signif_vst_data <- vst_data %>%
+```{r results_dataframe}
+# this is of class DESeqResults -- we want a data frame
+deseq_df <- deseq_shrunken %>%
+  # convert to a data frame
   as.data.frame() %>%
-  dplyr::filter(rownames(vst_data) %in% signif_genes)
+  # the gene ids were stored as row names -- let's them a column for easy display
+  tibble::rownames_to_column(var = "gene_id") %>%
+  # add on the gene names from the original deseq object
+  dplyr::mutate(gene_name = rowData(deseq_object)$gene_name)
 ```
 
-Let's make the results into a heatmap using the `pheatmap` package. 
-We will use the `scale = "row"` argument so that the data is standardized by gene (i.e., will be made into [z-scores](https://www.statisticshowto.datasciencecentral.com/probability-and-statistics/z-score/)). 
-Standardizing the data by gene makes it so our heatmap is more easily interpretable despite how different genes' distributions may be. 
-
-```{r pheatmap}
-pheatmap::pheatmap(signif_vst_data, 
-                   show_rownames = FALSE,
-                   scale = "row",
-                   colorRampPalette(c("deepskyblue",
-                                      "black",
-                                      "yellow"))(25))
+Let's print out the results table, sorted by log2 fold change. 
+The highest values should be genes more expressed in the MYCN amplified cell lines.
+
+```{r sorted_table, live = TRUE}
+# Print the table sorted by log2FoldChange 
+deseq_df %>% 
+  dplyr::arrange(dplyr::desc(log2FoldChange))
+```
+Now let's write the full results table to a file.
+
+```{r write_tsv}
+readr::write_tsv(deseq_df, file = deseq_df_file)
 ```
 
-If we want to save this heatmap as a PNG, we have to use the `png` function that is part of base R, run `pheatmap::pheatmap` again, and then use `dev.off()`.
-
-```{r png, live = TRUE}
-png(filename = heatmap_file)
-# Use the same command to create the heatmap
-pheatmap::pheatmap(signif_vst_data, 
-                   show_rownames = FALSE,
-                   scale = "row",
-                   colorRampPalette(c("deepskyblue",
-                                      "black",
-                                      "yellow"))(25))
-# This shuts down the current graphics device (the PNG we are saving)
-dev.off()
+
+#### Volcano Plot 
+
+With these shrunken effect sizes, we will draw a volcano plot, using the [`EnhancedVolcano` package](https://github.com/kevinblighe/EnhancedVolcano) to make it a bit easier.
+`EnhancedVolcano` has many, many options, which is a good thing if you don't like all of it's default settings.
```

Comment:
Are the defaults reasonable for most of our participants' use cases? (I think the answer is yes based on my little bit of knowledge of `EnhancedVolcano` but we may want to say so.)

Reply:
Which specific things do you think might be worth mentioning? I was thinking about maybe highlighting the cutoff values, but did you have other things in mind?

Reply:
> I was thinking about maybe highlighting the cutoff values, but did you have other things in mind?

I was thinking about the cutoffs as well.

Reply:
I added some detail, as well as comments about the options we use, in https://github.com/AlexsLemonade/training-modules/pull/405/commits/40c459b76fdc212083b8bed2363f0164c9b24b60